src/test.zig

```diff
@@ -51,3 +51,116 @@ fn testDislayContents(test_dir_path: []const u8, archive_name: []const u8, file_
         try testing.expect(mem.eql(u8, archive.files.items[index].name, file_name));
     }
 }
+
+test "test suite" {
+    var ctx = TestContext.init(std.heap.page_allocator);
+    //simple test: create a new gnu archive and compare the output //doesn't work for now 
+    {
+       
+        
+        var case = try ctx.addCase("create gnu archive from txt files", "test1");
+        
+        try case.args.append("zar");
+        try case.args.append("-rc");
+        try case.args.append("./test.a");
+        try case.args.append("./input1.txt");
+        try case.testOutput();
```

Comment:
Nothing is being done to add zar to path, so it will not work readily. It would be better to just call the zar's internal methods to run the tests

Reply:
no I have put in my /bin folder locally, the build script can be modified to put the zar binary there. I have tested all of this and it works.